sci-libs/pcl/pcl-9999.ebuild

```diff
@@ -69,37 +64,77 @@ REQUIRED_USE="
 "
 
 PATCHES=(
-	"${FILESDIR}"/${PN}-1.12.1-allow-configuration-of-install-dirs.patch
-	"${FILESDIR}"/${PN}-1.12.1-fix-hardcoded-relative-directory-of-the-installed-cmake-files.patch
+	"${FILESDIR}/${PN}-1.12.1-allow-configuration-of-install-dirs.patch"
+	"${FILESDIR}/${PN}-1.12.1-fix-hardcoded-relative-directory-of-the-installed-cmake-files.patch"
 )
 
-src_prepare() {
-	if use cuda; then
-		cuda_src_prepare
-		cuda_add_sandbox -w
-	fi
+cuda_set_CUDAHOSTCXX() {
```

Comment:
This is going to replace cuda_gccdir from [cuda.eclass](https://github.com/gentoo/gentoo/blob/master/eclass/cuda.eclass#L50). This does not rely on the hard-coded values from `cuda-config -s` and instead tests all installed compilers. It also is usable with both gcc and clang. This makes use of `crt/host_config.h` from `dev-util/nvidia-cuda-toolkit` and tests the actual failure we see and not `cuda-config` output. 

The variable name `CUDAHOSTCXX` comes from https://cmake.org/cmake/help/latest/envvar/CUDAHOSTCXX.html because all the places I used this for yet use CMake.

This solves the whole class of issues like https://bugs.gentoo.org/921246 and https://bugs.gentoo.org/913740 .

Reply:
Do we just need to match the major version of gcc and nvidia-cuda-toolkit? Or is it more complicated then that?